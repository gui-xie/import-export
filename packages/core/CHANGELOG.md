# @senlinz/import-export

## 0.0.1-beta.5

### Patch Changes

- fix
- Updated dependencies
  - @senlinz/import-export-wasm@0.0.1-beta.5

## 0.0.1-beta.4

### Patch Changes

- update to beta.4
- Updated dependencies
  - @senlinz/import-export-wasm@0.0.1-beta.4

## 0.0.1-beta.3

### Patch Changes

- update beta version
- Updated dependencies
  - @senlinz/import-export-wasm@0.0.1-beta.3

## 0.0.1-beta.2

### Patch Changes

- update version
- Updated dependencies
  - @senlinz/import-export-wasm@0.0.1-beta.2
