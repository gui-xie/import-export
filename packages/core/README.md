# @senlinz/import-export
example for import and export excel file

```html
<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>IMPORT-EXPORT-TEST</title>
</head>
<button id="btnImport">Import</button>
<button id="btnExport">Export</button>
<button id="btnDownloadTemplate">Download Template</button>
<div id="importOutput"></div>

<body>
  <script type="module">
    import { getUtils } from 'https://cdn.jsdelivr.net/npm/@senlinz/import-export@0.1.0-beta.7/+esm';
    const {
      importExcel,
      exportExcel,
      downloadExcelTemplate
    } = getUtils();

    const excelInfo = {
      name: 'TomAndJerry',
      sheetName: 'sheet1',
      columns: [
        { key: 'name', name: 'Name', dataType: 0, width: 20, note: 'required' },
        { key: 'age', name: 'Age', dataType: 1, width: 10 },
        { key: 'category', name: 'Category', dataType: 0, allowedValues: ['Cat', 'Mouse'] }
      ],
      author: 'senlinz',
      createTime: '2024-11-01T09:00:00'
    };

    btnImport.addEventListener('click', importClick);
    btnExport.addEventListener('click', exportClick);
    btnDownloadTemplate.addEventListener('click', downloadTemplateClick);

    function exportClick() {
      exportExcel(excelInfo, [
        { name: 'Tom', age: 12, category: 'Cat' },
        { name: 'Jerry', age: 13, category: 'Mouse' }
      ]);
    }

    function importClick() {
      importExcel(excelInfo).then(data => {
        importOutput.innerText = JSON.stringify(data, null);
      });
    }

    function downloadTemplateClick() {
      downloadExcelTemplate(excelInfo);
    }
  </script>
</body>

</html>
```