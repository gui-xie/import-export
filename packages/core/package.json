{
  "name": "@senlinz/import-export",
  "version": "0.0.1-beta.5",
  "description": "Stencil Component for import/export excel files.",
  "type": "module",
  "module": "dist/components/index.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist/**/*.js",
    "dist/**/*.d.ts",
    "README.md"
  ],
  "exports": {
    "./components": {
      "types": "./dist/types/index.d.ts"
    },
    "./components/imexport-table.js": {
      "import": "./dist/components/imexport-table.js",
      "types": "./dist/components/imexport-table.d.ts"
    }
  },
  "scripts": {
    "build": "stencil build",
    "start": "stencil build --dev --watch --serve",
    "test": "stencil test --spec",
    "e2e-serve": "serve -l 8080 ./tests",
    "e2e-test": "playwright test",
    "generate": "stencil generate",
    "publish-dry-run": "pnpm publish --dry-run --no-git-checks"
  },
  "devDependencies": {
    "@rollup/plugin-terser": "^0.4.4",
    "@stencil/core": "^4.7.0",
    "@stencil/vue-output-target": "^0.8.9",
    "@types/jest": "^29.5.6",
    "@types/node": "^16.18.108",
    "jest": "^29.7.0",
    "jest-cli": "^29.7.0"
  },
  "author": "senlin",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/gui-xie/import-export.git"
  },
  "dependencies": {
    "@senlinz/import-export-wasm": "workspace:*",
    "fflate": "^0.8.2"
  },
  "keywords": [
    "stencil",
    "webcomponents",
    "rust",
    "wasm",
    "import",
    "export",
    "excel",
    "xlsx",
    "sheet",
    "spreadsheet",
    "OpenXML",
    "browser",
    "client"
  ]
}