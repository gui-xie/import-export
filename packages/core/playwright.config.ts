import { PlaywrightTestConfig } from '@playwright/test';

const config: PlaywrightTestConfig = {
    use: {
        headless: false,
        baseURL: 'http://localhost:8080',
    },
    testDir: 'tests',
    snapshotDir: 'tests/snapshots',
    webServer: {
        command: 'npm run e2e-serve',
        port: 8080,
        timeout: 120 * 1000
    },
};

export default config;