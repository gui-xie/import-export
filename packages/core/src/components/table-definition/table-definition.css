.table-definition {
  margin: 20px;
}

.table-definition h2 {
  color: #333;
}

.table-definition h3 {
  color: #666;
}

.table-definition p {
  font-style: italic;
  color: #999;
}

.table-definition table {
  width: 100%;
  border-collapse: collapse;
  margin-top: 20px;
}

.table-definition th, .table-definition td {
  border: 1px solid #ddd;
  padding: 8px;
  text-align: left;
}

.table-definition th {
  background-color: #f4f4f4;
  color: #333;
}

.table-definition tr:nth-child(even) {
  background-color: #f9f9f9;
}

.table-definition tr:hover {
  background-color: #f1f1f1;
}