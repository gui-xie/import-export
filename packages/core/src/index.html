<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Document</title>
</head>

<body>
  <script type="module">
    import { getUtils } from './index.js';
    const {
      importExcel,
      exportExcel,
      downloadExcelTemplate
    } = getUtils();

    const excelInfo = {
      name: 'TomAndJerry',
      sheetName: 'sheet1',
      columns: [
        { key: 'name', name: 'Name', dataType: 0, width: 20, note: 'required' },
        { key: 'age', name: 'Age', dataType: 1, width: 4 },
        { key: 'category', name: 'Category', dataType: 1, allowedValues: ['Cat', 'Mouse'] }
      ]
    };

    importExcel(excelInfo).then(data => {
      console.log('imported data:', data);
    });
  </script>
</body>

</html>