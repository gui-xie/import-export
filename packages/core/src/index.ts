export type * from './components.d.ts';
import { importExcel, exportExcel, downloadExcelTemplate, fromExcel, toExcel, generateExcelTemplate, initializeWasm } from './utils/utils';

function getUtils() {
    initializeWasm();
    return {
        importExcel,
        exportExcel,
        downloadExcelTemplate,
        fromExcel,
        toExcel,
        generateExcelTemplate
    }
}

export { getUtils };