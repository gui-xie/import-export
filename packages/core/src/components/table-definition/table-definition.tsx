import { Component, Prop, h } from '@stencil/core';
import { ExcelDefinition } from '../../declarations/ExcelDefintion';

@Component({
  tag: 'table-definition',
  styleUrl: 'table-definition.css',
  shadow: true,
})
export class TableDefinition {
  @Prop() definition: ExcelDefinition;

  render() {
    return this.definition ? (
      <div class="table-definition">
        <h2 class="definition-name">{this.definition.name}</h2>
        <h3>Sheet: {this.definition.sheetName}</h3>
        {this.definition.author && <p>Author: {this.definition.author}</p>}
        <table>
          <thead>
            <tr>
              <th>Key</th>
              <th>Name</th>
              <th>Width</th>
              <th>Note</th>
              <th>Data Type</th>
              <th>Allowed Values</th>
            </tr>
          </thead>
          <tbody>
            {this.definition.columns.map(column => (
              <tr>
                <td>{column.key}</td>
                <td>{column.name}</td>
                <td>{column.width ?? 'N/A'}</td>
                <td>{column.note ?? 'N/A'}</td>
                <td>{column.dataType ?? 'N/A'}</td>
                <td>{column.allowedValues?.join(', ') ?? 'N/A'}</td>
              </tr>
            ))}
          </tbody>
        </table>
      </div>
    ) : <div class="table-definition">No definition provided</div>;
  }
}