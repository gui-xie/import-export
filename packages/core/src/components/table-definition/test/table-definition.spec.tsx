import { newSpecPage } from '@stencil/core/testing';
import { TableDefinition } from '../table-definition';

describe('table-definition', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [TableDefinition],
      html: `<table-definition></table-definition>`,
    });
    const tableDefinition = page.body.querySelector('table-definition');
    tableDefinition.definition = {
      name: 'TomAndJerry',
      sheetName: 'Sheet1',
      columns: [
        {
          key: 'name',
          name: 'Name',
        },
        {
          key: 'birth',
          name: 'Birth',
        },
      ],
    };

    await page.waitForChanges();

    expect(tableDefinition).toMatchSnapshot();
  });
});
