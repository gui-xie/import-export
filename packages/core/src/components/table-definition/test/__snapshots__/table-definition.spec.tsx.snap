// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`table-definition renders 1`] = `
<table-definition>
  <template shadowrootmode="open">
    <div class="table-definition">
      <h2 class="definition-name">
        TomAndJerry
      </h2>
      <h3>
        Sheet: Sheet1
      </h3>
      <table>
        <thead>
          <tr>
            <th>
              Key
            </th>
            <th>
              Name
            </th>
            <th>
              Width
            </th>
            <th>
              Note
            </th>
            <th>
              Data Type
            </th>
            <th>
              Allowed Values
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              name
            </td>
            <td>
              Name
            </td>
            <td>
              N/A
            </td>
            <td>
              N/A
            </td>
            <td>
              N/A
            </td>
            <td>
              N/A
            </td>
          </tr>
          <tr>
            <td>
              birth
            </td>
            <td>
              Birth
            </td>
            <td>
              N/A
            </td>
            <td>
              N/A
            </td>
            <td>
              N/A
            </td>
            <td>
              N/A
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </template>
</table-definition>
`;
